galley/pkg/server/components/processing2.go

```diff
@@ -0,0 +1,346 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package components
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+
+	mcp "istio.io/api/mcp/v1alpha1"
+	"istio.io/istio/galley/pkg/config/event"
+	"istio.io/istio/galley/pkg/config/processing"
+	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/config/schema"
+	"istio.io/istio/galley/pkg/config/source/kube"
+	"istio.io/istio/galley/pkg/config/source/kube/apiserver"
+	"istio.io/istio/galley/pkg/config/source/kube/rt"
+	"istio.io/istio/galley/pkg/runtime/groups"
+	"istio.io/istio/galley/pkg/server/process"
+	"istio.io/istio/galley/pkg/server/settings"
+	configz "istio.io/istio/pkg/mcp/configz/server"
+	"istio.io/istio/pkg/mcp/creds"
+	"istio.io/istio/pkg/mcp/monitoring"
+	mcprate "istio.io/istio/pkg/mcp/rate"
+	"istio.io/istio/pkg/mcp/server"
+	"istio.io/istio/pkg/mcp/snapshot"
+	"istio.io/istio/pkg/mcp/source"
+	"istio.io/pkg/ctrlz/fw"
+	"istio.io/pkg/log"
+)
+
+// Processing2 component.
+type Processing2 struct {
```

Comment:
Processing2? It seems like this is essentially boostrap code for galley ... maybe a replacement for the server, no? Need a better comment to make it clear what this is and how it fits into galley.

Reply:
It is similar to other components. I've refactored server.go some time ago in master to disentangle the mess of subsystems. This is one of many.

I'll add some more comments in a later check-in to clarify.